packages/amplify-ui-components/src/components/amplify-s3-image/amplify-s3-image.tsx

```diff
@@ -0,0 +1,101 @@
+import { Component, Element, Prop, h, State, Host } from '@stencil/core';
+import { image } from './amplify-s3-image.style';
+import { styleNuker } from '../../common/helpers';
+import { AMPLIFY_UI_PREFIX, NO_STORAGE_MODULE_FOUND } from '../../common/constants';
+import { Storage } from '@aws-amplify/storage';
+import { Logger } from '@aws-amplify/core';
+
+const logger = new Logger('S3Image');
+
+const STATIC_LINK_CLASS_NAME = `${AMPLIFY_UI_PREFIX}--s3-image`;
+
+@Component({
+  tag: 'amplify-s3-image',
+  shadow: false,
+})
+export class AmplifyS3Image {
+  @Element() el: HTMLElement;
+
+  /* The key of the image object in S3 */
+  @Prop() imgKey: string;
+  /* String representing directory location to image file */
+  @Prop() path: string;
+  /* Image body content to be uploaded */
+  @Prop() body: object;
+  /* The content type header used when uploading to S3 */
+  @Prop() contentType: string = 'binary/octet-stream';
+  /* The accessbility level of the image */
+  @Prop() level: string;
+  /* Whether or not to use track the get/put of the image */
+  @Prop() track: boolean;
+  /* Cognito ideneity id of the another user's image */
+  @Prop() identityId: string;
+  /* Whether or not the photo picker is enabled */
+  @Prop() pickerEnabled: boolean = true;
+  /** Override default styling */
+  @Prop() overrideStyle: boolean = false;
+  /* Function executed when image loads */
+  @Prop() handleOnLoad: () => {};
+  /* Function executed when error occurs for the image */
+  @Prop() handleOnError: () => {};
+
+  @State() src: string | Object = null;
+
+  async componentWillLoad() {
+    await this.load();
+  }
+
+  async load() {
+    const { imgKey, path, body, contentType, level, track, identityId } = this;
+    if (!imgKey && !path) {
+      logger.debug('empty imgKey and path');
+      return;
+    }
+
+    const key = imgKey || path;
+    logger.debug('loading ' + key + '...');
+    if (body) {
+      if (!Storage || typeof Storage.put !== 'function') {
+        throw new Error(NO_STORAGE_MODULE_FOUND);
+      }
+
+      try {
+        const data = Storage.put(key, body, {
+          contentType,
+          level: level ? level : 'public',
+          track,
+        });
+        logger.debug(data);
+        this.src = await this.getImageSource(key, level, track, identityId);
+      } catch (error) {
+        logger.error(error);
+        throw new Error(error);
+      }
+    } else {
+      this.src = await this.getImageSource(key, level, track, identityId);
+    }
+  }
+
+  async getImageSource(key, level, track, identityId) {
+    if (!Storage || typeof Storage.get !== 'function') {
+      throw new Error(NO_STORAGE_MODULE_FOUND);
+    }
+
+    try {
+      const src = await Storage.get(key, { level: level ? level : 'public', track, identityId });
+      return src;
```

Comment:
Would the below suggestion work? 
```suggestion
      await Storage.get(key, { level: level ? level : 'public', track, identityId });
      return;
```

I guess either is fine cause ES linter recommends both so

Reply:
The suggestion would not work because the anything consuming `getImageSource` expects it to return a string